packages/flutter_tools/lib/src/drive/web_driver_service.dart

```diff
@@ -0,0 +1,273 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:file/file.dart';
+import 'package:meta/meta.dart';
+import 'package:webdriver/async_io.dart' as async_io;
+
+import '../base/common.dart';
+import '../base/process.dart';
+import '../build_info.dart';
+import '../convert.dart';
+import '../device.dart';
+import '../globals.dart' as globals;
+import '../project.dart';
+import '../resident_runner.dart';
+import '../web/web_runner.dart';
+import 'drive_service.dart';
+
+/// An implementation of the driver service for web debug and release applications.
+class WebDriverService extends DriverService {
+  WebDriverService({
+    @required ProcessUtils processUtils,
+    @required String dartSdkPath,
+  }) : _processUtils = processUtils,
+       _dartSdkPath = dartSdkPath;
+
+  final ProcessUtils _processUtils;
+  final String _dartSdkPath;
+
+  ResidentRunner _residentRunner;
+  Uri _webUri;
+
+  @override
+  Future<void> start(
+    BuildInfo buildInfo,
+    Device device,
+    DebuggingOptions debuggingOptions,
+    bool ipv6, {
+    File applicationBinary,
+    String route,
+    String userIdentifier,
+    String mainPath,
+    Map<String, Object> platformArgs = const <String, Object>{},
+  }) async {
+    final FlutterDevice flutterDevice = await FlutterDevice.create(
+      device,
+      target: mainPath,
+      buildInfo: buildInfo,
+      platform: globals.platform,
+    );
+    _residentRunner = webRunnerFactory.createWebRunner(
+      flutterDevice,
+      target: mainPath,
+      ipv6: ipv6,
+      debuggingOptions: buildInfo.isRelease ?
```

Comment:
This is the fix. for mobile/desktop we default to start paused which does not yet work for the web driver code.

Reply:
How about other options? Is using the default value the right choice?

Reply:
it is consistent with the previous behavior. I think supporting more flags is valuable but out of scope for this change